{
   "status" : "OK",
   "result" : [
      {
         "workSourceName" : "Default Snapshot",
         "user" : null,
         "actionType" : "DB_SYNC",
         "title" : "DB_SYNC",
         "details" : "Run SnapSync for database \"si4rac\".",
         "failureAction" : "Make sure host is available and the SSH server/service is running.",
         "reference" : "ACTION-9557",
         "namespace" : null,
         "endTime" : "2017-10-12T15:38:13.708Z",
         "state" : "FAILED",
         "workSource" : "POLICY",
         "failureDescription" : "The host \"[target3]\" is unavailable.",
         "report" : null,
         "failureMessageCode" : "exception.oracle.accessor.host.unavailable",
         "startTime" : "2017-10-12T15:38:02.658Z",
         "type" : "Action",
         "parentAction" : null,
         "userAgent" : null
      },
      {
         "userAgent" : null,
         "startTime" : "2017-10-11T16:42:09.795Z",
         "type" : "Action",
         "parentAction" : "ACTION-9545",
         "failureMessageCode" : "exception.oracle.dsource.sync.no_hosts.rac",
         "report" : null,
         "workSource" : "WEBSERVICE",
         "state" : "FAILED",
         "failureDescription" : "Delphix Engine cannot connect to any host to load data from source \"racdba\": SnapSync could not retrieve runtime information for database \"racdba\".",
         "endTime" : "2017-10-11T16:42:43.486Z",
         "reference" : "ACTION-9548",
         "namespace" : null,
         "details" : "Run SnapSync for database \"racdba\".",
         "failureAction" : "Check database accessibility and try again.",
         "title" : "DB_SYNC",
         "workSourceName" : "delphix_admin",
         "user" : "USER-2",
         "actionType" : "DB_SYNC"
      },
      {
         "state" : "FAILED",
         "workSource" : "WEBSERVICE",
         "failureDescription" : "Cannot SnapSync database \"racdba\" because database is not in ARCHIVELOG mode.",
         "report" : null,
         "parentAction" : null,
         "type" : "Action",
         "startTime" : "2017-10-11T16:16:12.174Z",
         "userAgent" : "Delphix Perl Agent/0.1",
         "failureMessageCode" : "exception.oracle.dsource.sync.noarchivelog",
         "actionType" : "DB_SYNC",
         "user" : "USER-2",
         "workSourceName" : "delphix_admin",
         "reference" : "ACTION-9530",
         "namespace" : null,
         "endTime" : "2017-10-11T16:16:46.144Z",
         "title" : "DB_SYNC",
         "failureAction" : "Perform SnapSync when database is in ARCHIVELOG mode.",
         "details" : "Run SnapSync for database \"racdba\"."
      },
      {
         "workSource" : "WEBSERVICE",
         "failureDescription" : "Connection to database \"racdba\" with the given credential for user \"delphixdb\" was denied.",
         "state" : "FAILED",
         "report" : null,
         "startTime" : "2017-10-11T16:13:45.836Z",
         "type" : "Action",
         "parentAction" : null,
         "userAgent" : "Delphix Perl Agent/0.1",
         "failureMessageCode" : "exception.oracle.dbc.credentials.invalid",
         "workSourceName" : "delphix_admin",
         "actionType" : "DB_SYNC",
         "user" : "USER-2",
         "namespace" : null,
         "reference" : "ACTION-9528",
         "endTime" : "2017-10-11T16:13:50.748Z",
         "title" : "DB_SYNC",
         "details" : "Run SnapSync for database \"racdba\".",
         "failureAction" : "Make sure that the configured database username and password is correct."
      },
      {
         "workSourceName" : "delphix_admin",
         "actionType" : "SNAPSHOT_DELETE",
         "user" : "USER-2",
         "title" : "SNAPSHOT_DELETE",
         "details" : "Delete snapshot \"@2017-10-10T08:39:04.451Z\".",
         "failureAction" : "Delete the database \"Analytics/si4rac\" if the only associated snapshot is not needed.",
         "reference" : "ACTION-9494",
         "namespace" : null,
         "endTime" : "2017-10-10T11:52:02.734Z",
         "workSource" : "WEBSERVICE",
         "failureDescription" : "Cannot remove a snapshot when it is the only snapshot for the container.",
         "state" : "FAILED",
         "report" : null,
         "failureMessageCode" : "exception.db.genericcontainer.cannot.remove.only.snapshot",
         "type" : "Action",
         "startTime" : "2017-10-10T11:52:02.695Z",
         "parentAction" : null,
         "userAgent" : "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36"
      },
      {
         "namespace" : null,
         "reference" : "ACTION-9488",
         "endTime" : "2017-10-10T11:51:23.833Z",
         "title" : "SNAPSHOT_DELETE",
         "details" : "Delete snapshot \"@2017-10-10T08:39:04.451Z\".",
         "failureAction" : "Delete the dependent datasets and try again.",
         "workSourceName" : "delphix_admin",
         "actionType" : "SNAPSHOT_DELETE",
         "user" : "USER-2",
         "startTime" : "2017-10-10T11:51:23.712Z",
         "type" : "Action",
         "parentAction" : null,
         "userAgent" : "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36",
         "failureMessageCode" : "exception.storage.dependents",
         "failureDescription" : "Could not delete Analytics/si4rac@2017-10-10T08:39:04.451Z because the following dependent datasets exist:\n\nAnalytics/snapsize",
         "workSource" : "WEBSERVICE",
         "state" : "FAILED",
         "report" : null
      },
      {
         "report" : null,
         "failureDescription" : "Could not delete Sources/Oracle dsource@2017-09-28T10:21:08.908Z because the following dependent datasets exist:\n\nAnalytics/autotest\nAnalytics/depend",
         "state" : "FAILED",
         "workSource" : "WEBSERVICE",
         "failureMessageCode" : "exception.storage.dependents",
         "userAgent" : "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36",
         "startTime" : "2017-10-10T09:50:43.910Z",
         "type" : "Action",
         "parentAction" : null,
         "workSourceName" : "delphix_admin",
         "actionType" : "SNAPSHOT_DELETE",
         "user" : "USER-2",
         "details" : "Delete snapshot \"@2017-09-28T10:21:08.908Z\".",
         "failureAction" : "Delete the dependent datasets and try again.",
         "title" : "SNAPSHOT_DELETE",
         "endTime" : "2017-10-10T09:50:44.026Z",
         "reference" : "ACTION-9482",
         "namespace" : null
      },
      {
         "failureMessageCode" : "exception.storage.dependents",
         "userAgent" : "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36",
         "parentAction" : null,
         "startTime" : "2017-10-10T09:49:29.551Z",
         "type" : "Action",
         "report" : null,
         "state" : "FAILED",
         "workSource" : "WEBSERVICE",
         "failureDescription" : "Could not delete Sources/Oracle dsource@2017-09-28T10:21:08.908Z because the following dependent datasets exist:\n\nAnalytics/autotest\nAnalytics/depend",
         "failureAction" : "Delete the dependent datasets and try again.",
         "details" : "Delete snapshot \"@2017-09-28T10:21:08.908Z\".",
         "title" : "SNAPSHOT_DELETE",
         "endTime" : "2017-10-10T09:49:29.639Z",
         "reference" : "ACTION-9481",
         "namespace" : null,
         "actionType" : "SNAPSHOT_DELETE",
         "user" : "USER-2",
         "workSourceName" : "delphix_admin"
      },
      {
         "userAgent" : "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36",
         "type" : "Action",
         "startTime" : "2017-10-10T08:38:29.259Z",
         "parentAction" : null,
         "failureMessageCode" : "exception.oracle.dbc.database.failed.connection",
         "report" : null,
         "workSource" : "WEBSERVICE",
         "failureDescription" : "Cannot connect to database due to: Exception occurred while getting connection: oracle.ucp.UniversalConnectionPoolException: Cannot get Connection from Datasource: java.sql.SQLException: ORA-28001: the password has expired\n",
         "state" : "FAILED",
         "endTime" : "2017-10-10T08:38:30.816Z",
         "namespace" : null,
         "reference" : "ACTION-9457",
         "details" : "Run SnapSync for database \"si4rac\".",
         "failureAction" : "Make sure that the username provided has access rights to system views, and that the host server can be reached.",
         "title" : "DB_SYNC",
         "workSourceName" : "delphix_admin",
         "user" : "USER-2",
         "actionType" : "DB_SYNC"
      },
      {
         "failureMessageCode" : "exception.oracle.dbc.database.failed.connection",
         "userAgent" : "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36",
         "type" : "Action",
         "startTime" : "2017-10-10T08:37:32.780Z",
         "parentAction" : null,
         "report" : null,
         "state" : "FAILED",
         "failureDescription" : "Cannot connect to database due to: Exception occurred while getting connection: oracle.ucp.UniversalConnectionPoolException: Cannot get Connection from Datasource: java.sql.SQLRecoverableException: IO Error: The Network Adapter could not establish the connection",
         "workSource" : "WEBSERVICE",
         "details" : "Run SnapSync for database \"si4rac\".",
         "failureAction" : "Make sure that the username provided has access rights to system views, and that the host server can be reached.",
         "title" : "DB_SYNC",
         "endTime" : "2017-10-10T08:37:34.076Z",
         "namespace" : null,
         "reference" : "ACTION-9456",
         "workSourceName" : "delphix_admin",
         "user" : "USER-2",
         "actionType" : "DB_SYNC"
      },
      {
         "actionType" : "DB_DELETE",
         "user" : "USER-2",
         "workSourceName" : "delphix_admin",
         "endTime" : "2017-10-10T08:35:33.782Z",
         "reference" : "ACTION-9453",
         "namespace" : null,
         "failureAction" : "Make sure that the device belongs to the Delphix target, or drop the device and try the operation again.",
         "details" : "Delete dataset \"tests\".",
         "title" : "DB_DELETE",
         "report" : null,
         "failureDescription" : "Device \"target_v2p\" for target database \"tests\" on environment \"LINUXTARGET\" and instance \"LINUXTARGET\" is not on the Delphix mount path \"/u01/app/toolkit/564d870b-e091-7339-5d98-e7cc4bdbfbcc-vdb-46/datafile\".",
         "workSource" : "WEBSERVICE",
         "state" : "FAILED",
         "userAgent" : "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36",
         "parentAction" : null,
         "startTime" : "2017-10-10T08:35:33.658Z",
         "type" : "Action",
         "failureMessageCode" : "exception.db.asetarget.target.device.not_on_delphix"
      },
      {
         "failureAction" : "Make sure that the device belongs to the Delphix target, or drop the device and try the operation again.",
         "details" : "Stop dataset \"tests\".",
         "title" : "SOURCE_STOP",
         "endTime" : "2017-10-10T08:35:03.795Z",
         "reference" : "ACTION-9451",
         "namespace" : null,
         "user" : "USER-2",
         "actionType" : "SOURCE_STOP",
         "workSourceName" : "delphix_admin",
         "failureMessageCode" : "exception.db.asetarget.target.device.not_on_delphix",
         "userAgent" : "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36",
         "parentAction" : null,
         "startTime" : "2017-10-10T08:35:03.437Z",
         "type" : "Action",
         "report" : null,
         "failureDescription" : "Device \"target_v2p\" for target database \"tests\" on environment \"LINUXTARGET\" and instance \"LINUXTARGET\" is not on the Delphix mount path \"/u01/app/toolkit/564d870b-e091-7339-5d98-e7cc4bdbfbcc-vdb-46/datafile\".",
         "state" : "FAILED",
         "workSource" : "WEBSERVICE"
      },
      {
         "workSourceName" : "delphix_admin",
         "actionType" : "DB_SYNC",
         "user" : "USER-2",
         "endTime" : "2017-10-10T08:34:55.488Z",
         "reference" : "ACTION-9450",
         "namespace" : null,
         "details" : "Run SnapSync for database \"tests\".",
         "failureAction" : "Contact Delphix Support.",
         "title" : "DB_SYNC",
         "report" : null,
         "state" : "FAILED",
         "workSource" : "WEBSERVICE",
         "failureDescription" : "DB_SYNC job for \"Analytics/tests\" failed due to an internal error during execution.",
         "userAgent" : "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36",
         "startTime" : "2017-10-10T08:34:55.231Z",
         "type" : "Action",
         "parentAction" : null,
         "failureMessageCode" : "exception.job.execute.internal"
      },
      {
         "namespace" : null,
         "reference" : "ACTION-9419",
         "endTime" : "2017-10-09T08:39:40.276Z",
         "title" : "DB_SYNC",
         "failureAction" : "Make sure the instance is accepting remote connections over TCP, the instance is not using a dynamic TCP port, and the SQL login \"sa\" has privileges to connect to the instance. Then refresh the environment and try again.",
         "details" : "Run SnapSync for database \"tests\".",
         "actionType" : "DB_SYNC",
         "user" : null,
         "workSourceName" : "Default Snapshot",
         "parentAction" : null,
         "startTime" : "2017-10-09T08:37:54.785Z",
         "type" : "Action",
         "userAgent" : null,
         "failureMessageCode" : "exception.db.ase.instance.jdbc_query.failed",
         "workSource" : "POLICY",
         "state" : "FAILED",
         "failureDescription" : "Failed to connect to instance \"LINUXTARGET\" over JDBC.",
         "report" : null
      }
   ],
   "type" : "ListResult",
   "total" : 14,
   "action" : null,
   "overflow" : false,
   "job" : null
}
